### PR TITLE
Tuple type impl

### PR DESCRIPTION
**Pull request by radkomih**
**Detailed description**:

**Which issue(s) this PR fixes**:
Fixes [#5](https://github.com/LimeChain/goscale/issues/5)

**Special notes for your reviewer**:

**Checklist**
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: sequence.go [hunk 1]
```diff
@@ -9,14 +9,20 @@ package goscale
...
 )
 
 type Sequence[T Encodable] []T
 
 func (seq Sequence[Encodable]) Encode(buffer *bytes.Buffer) {
 	Compact(len(seq)).Encode(buffer)
+
 	for _, v := range seq {
-		v.Encode(buffer)
+		if reflect.TypeOf(v).Kind() != reflect.Struct {
```
Comment (failfmi) [inline sequence.go:21]: Could we revert the switch? I think it would be more readable.

Comment (radkomih) [reply sequence.go:21]: 👍

File: tuple.go [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+			case reflect.Uintptr:
+				panic("encoding of Uintptr field is not implemented")
+			case reflect.UnsafePointer:
+				panic("encoding of UnsafePointer field is not implemented")
+			case reflect.Pointer:
+				panic("encoding of Pointer field is not implemented")
+			case reflect.Chan:
+				panic("encoding of Chan field is not implemented")
+			case reflect.Func:
+				panic("encoding of Func field is not implemented")
```
Comment (failfmi) [inline tuple.go:116]: nit: we can match those in one case with `panic("not implemented")`

File: tuple.go [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+
+	// TODO: Sequence[Sequence[T]]
+	// TODO: Sequence[FixedSequence[T]]
+	// TODO: Sequence[Dictionary[T1, T2]]
+
+	default:
+
+		switch field.Type() {
+		case reflect.TypeOf(*new(VaryingData)):
+			ConvertTo[VaryingData](field).Encode(buffer)
```
Comment (failfmi) [inline tuple.go:173]: Is this needed?

**Review by failfmi — APPROVED** [review]

File: tuple.go (hunk 2 above)
Comment (radkomih) [reply tuple.go:116]: 👍

File: tuple.go (hunk 3 above)
Comment (radkomih) [reply tuple.go:173]: yes, here  `func Test_EncodeTupleVaryingData(t *testing.T) {`

**Review by failfmi — APPROVED** [review]

**Pull request merged**